src/pages/api/signUp.ts

```diff
@@ -0,0 +1,29 @@
+import bcrypt from 'bcrypt'
+import connectDB from "@/lib/mongodb";
+import User from "@/types/User";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    if (req.method !== 'POST') {
+        return res.status(405).end()
+    }
+    const { name, email, password, rePassword } = req.body
+    if (password !== rePassword) {
+        return res.status(400).json({ error: 'Password miss match' })
+    }
+
+    await connectDB()
+    const isUserExist = await User.findOne({ email })
```

Comment:
not a valid english expression it should be `doesUserExists` or just `user` because this does not return you boolean but the object.